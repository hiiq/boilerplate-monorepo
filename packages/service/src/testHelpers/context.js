import { User, Utils } from '@boilerplate-monorepo/common';

const stubLoader = name => ({
  clear: td.func(`${name}.clear`),
  load: td.func(`${name}.load`),
});

const db = {
  user: {
    read: jest.fn().mockName('read'),
    readRaw: jest.fn().mockName('readRaw'),
    recoveryFind: jest.fn().mockName('recoveryFind'),
    register: jest.fn().mockName('register'),
    revokeRefreshTokens: jest.fn().mockName('revokeRefreshTokens'),
    save: jest.fn().mockName('save'),
  },
  users: {
    read: jest.fn().mockName('read'),
    readPaginated: jest.fn().mockName('readPaginated'),
    readRaw: jest.fn().mockName('readRaw'),
  },
};

const make = Utils.makeSafeExample({
  db,
  ip: '127.0.0.1',
  loaders: {
    user: stubLoader('user'),
  },
  req: {
    headers: {},
    ip: '127.0.0.1',
  },
  res: {
    cookie: jest.fn().mockName('cookie'),
  },
  user: User.apiExample(),
});

export { make };
