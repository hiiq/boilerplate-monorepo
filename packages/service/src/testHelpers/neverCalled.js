const neverCalled = { ignoreExtraArgs: true, times: 0 };

export { neverCalled };
