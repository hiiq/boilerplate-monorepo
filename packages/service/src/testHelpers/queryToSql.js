import { pick } from 'ramda';

const queryToSql = pick(['bindings', 'sql', 'transaction']);

export { queryToSql };
