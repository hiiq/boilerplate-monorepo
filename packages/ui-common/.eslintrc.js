const path = require('path');

module.exports = {
  env: {
    browser: true,
    jest: true,
  },
  plugins: ['eslint-comments', 'import', 'jest', 'import-helpers', 'ramda'],
  root: false,
  settings: {
    'import/resolver': {
      node: {
        paths: [path.resolve(__dirname, 'src')],
      },
    },
  },
};
