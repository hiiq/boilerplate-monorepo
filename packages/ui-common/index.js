export { A11y } from './src/a11y';
