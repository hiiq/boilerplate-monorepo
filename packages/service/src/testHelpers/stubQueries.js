import knexMock from 'mock-knex';
import { queryToSql } from './queryToSql';

const stubQueries = (mutateFn, responses = []) => {
  const tracker = knexMock.getTracker();

  tracker.install();

  let queries = [];

  tracker.on('query', (query, step) => {
    queries = [...queries, queryToSql(query)];
    mutateFn(queries);
    query.response(responses[step - 1]);
  });
};

export { stubQueries };
