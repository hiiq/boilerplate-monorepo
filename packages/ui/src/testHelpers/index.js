import { Theme, Modality } from '@boilerplate-monorepo/ui-common';
import { render } from '@testing-library/react';
import { node } from 'prop-types';
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { ThemeProvider } from 'styled-components/macro';
import { Provider } from 'types/provider';

export * from '@testing-library/react';

jest.mock('shared/useTheme');
jest.mock('shared/useTranslate');

const AllTheProviders = ({ children }) => (
  <Router>
    <Provider.MODALITY value={Modality.initial}>
      <ThemeProvider theme={Theme.example()}>{children}</ThemeProvider>
    </Provider.MODALITY>
  </Router>
);

AllTheProviders.propTypes = {
  children: node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export { customRender as render };
