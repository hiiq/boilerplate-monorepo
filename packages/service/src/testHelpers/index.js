import { make as makeContext } from './context';
import {
  graphqlClient,
  makeGraphqlClient,
  responseToData,
  responseToErrorCode,
} from './graphql';
import { neverCalled } from './neverCalled';
import { queryToSql } from './queryToSql';
import { stubQueries } from './stubQueries';

export {
  graphqlClient,
  makeContext,
  makeGraphqlClient,
  neverCalled,
  queryToSql,
  responseToData,
  responseToErrorCode,
  stubQueries,
};
