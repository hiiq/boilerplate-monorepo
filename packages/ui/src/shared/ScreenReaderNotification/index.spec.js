import { A11y } from '@boilerplate-monorepo/ui-common';
import React from 'react';
import { render } from 'testHelpers';
import { ScreenReaderNotification } from '.';

const { Politeness } = A11y;

describe('<ScreenReaderNotification/>', () => {
  const politeness = Politeness.POLITE;
  const defaultProps = {
    children: 'CHILDREN',
  };

  const renderComponent = (overrides) =>
    render(<ScreenReaderNotification {...defaultProps} {...overrides} />);

  test('renders children', () => {
    const { container } = renderComponent();

    expect(container.firstChild).toMatchSnapshot();
  });

  test('it has the proper aria attributes', () => {
    const { container } = renderComponent();

    expect(container.firstChild).toHaveAttribute('aria-live', politeness);
  });
});
