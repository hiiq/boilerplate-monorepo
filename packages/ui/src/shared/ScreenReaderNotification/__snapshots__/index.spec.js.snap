// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScreenReaderNotification/> renders children 1`] = `
<div
  aria-live="polite"
  class="ScreenReaderNotification__Container-jzwqtd-0 huOtLM"
>
  CHILDREN
</div>
`;
